### PR TITLE
Feature/homework3

### PR DESCRIPTION
**Pull request by Secondtwin**

### CODE REVIEW

File: hw2/models/user.schema.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as Joi from 'joi';
 
 export const userSchema = Joi.object({
-    id: Joi.string().required(),
+    id: Joi.string(),
```
Comment (vzhufk) [inline hw2/models/user.schema.ts:4]: You are adding the `id` field into the input schema, which means I can change my user id via request, which should not be allowed.

File: hw2/api/routing.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+let dataAccessService: UserDataService;
+let userService: UserService;
+
+export function initialize(sequelize: Sequelize): void {
+    dataAccessService = new UserDataService(sequelize);
+    userService = new UserService(dataAccessService);
+}
+
+router.get('/getUserList', (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:19]: To follow REST.
```
router.get('/users', (req: Request, res: Response) => {
```

File: hw2/api/routing.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const users = userService.getUserList();
+
+    res.json(users);
+});
+
+router.get('/getUser/:id', (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:25]: To follow REST.
```
router.get('/users/:id', (req: Request, res: Response) => {
```

File: hw2/api/routing.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const user = userService.getUserById(req.params.id);
+
+    res.json(user);
+});
+
+router.delete('/deleteUser/:id', (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:31]: To follow REST.
```
router.delete('/users/:id', (req: Request, res: Response) => {
````

File: hw2/api/routing.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const user = userService.deleteUserById(req.params.id);
+
+    res.json(user);
+});
+
+router.post('/addUser', validateUser(userSchema), (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:37]: To follow REST:
```
router.post('/users', validateUser(userSchema), (req: Request, res: Response) => {
```

File: hw2/api/routing.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    const user = userService.addUser(req.body);
+
+    res.json(user);
+});
+
+router.put('/updateUser/:id', validateUser(userSchema), (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:43]: To follow REST:
```
router.put('/users/:id', validateUser(userSchema), (req: Request, res: Response) => {
```

File: hw2/api/routing.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    const user = userService.updateUserById(req.params.id, req.body);
+
+    res.json(user);
+});
+
+router.get('/getAutoSuggestUsers', (req: Request, res: Response) => {
```
Comment (vzhufk) [inline hw2/api/routing.ts:49]: To follow REST:
```
router.get('/users/auto-suggested', (req: Request, res: Response) => {
```

File: hw2/config/enviroment.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export const environment = {
+    databaseName: 'mentor_prgr_hw3',
```
Comment (vzhufk) [inline hw2/config/enviroment.ts:2]: Also use `process.env` here.

File: hw2/models/user.schema.ts (hunk 1 above)
Comment (vzhufk) [reply hw2/models/user.schema.ts:4]: The body probably should not include the `id` field.

**Comment by Secondtwin** [comment]
@vzhufk fixes performed in PR: https://github.com/Secondtwin/NodeJS_Mentor_Program/pull/8

**Pull request merged**